### PR TITLE
Make complement for ImmutableConciseSet faster

### PR DESCRIPTION
**Pull request by drcrallen**

### CODE REVIEW

File: src/test/java/it.uniroma3.mat.extendedset/intset/ImmutableConciseSetTest.java [hunk 1]
```diff
@@ -1823,6 +1823,116 @@ public void testComplement9()
     verifyComplement(expected, testSet, length);
   }
 
+
+  /**
+   * Complement of a null set with a length
+   */
+  @Test
+  public void testComplement9_31()
```
Comment (xvrl) [inline src/test/java/it.uniroma3.mat.extendedset/intset/ImmutableConciseSetTest.java]: Maybe we can combine all your tests into one with a set of specific lengths we'd like to test, and then also add a bunch of randomly generated lengths between 0 and Integer.MAX_VALUE?

Comment (drcrallen) [inline src/test/java/it.uniroma3.mat.extendedset/intset/ImmutableConciseSetTest.java]: Fixed. Also added tests up to the limited number supported

File: src/test/java/it.uniroma3.mat.extendedset/intset/ImmutableConciseSetTest.java [hunk 2]
```diff
@@ -1811,16 +1815,43 @@ public void testComplement8()
...
+            1,
+            0,
+            31 * 3,
+            1024,
+            ConciseSetUtils.MAX_ALLOWED_INTEGER
+        )
+    );
+    final Random random = new Random(701534702L);
+    for (int i = 0; i < 10; ++i) {
+      lengths.add(Math.abs(random.nextInt(ConciseSetUtils.MAX_ALLOWED_INTEGER + 1)));
```
Comment (xvrl) [inline src/test/java/it.uniroma3.mat.extendedset/intset/ImmutableConciseSetTest.java]: nextInt is always positive

File: src/test/java/it.uniroma3.mat.extendedset/intset/ImmutableConciseSetTest.java [hunk 3]
```diff
@@ -1811,16 +1815,43 @@ public void testComplement8()
...
+    }
+    final ImmutableConciseSet emptySet = new ImmutableConciseSet();
+    Collections.shuffle(lengths, random);
```
Comment (xvrl) [inline src/test/java/it.uniroma3.mat.extendedset/intset/ImmutableConciseSetTest.java]: any reason we are shuffling the list of lenghts?

**Pull request merged**